src/main/java/org/sonatype/repository/conan/internal/proxy/ConanProxyFacet.java

```diff
@@ -130,33 +140,50 @@ private Content putPackage(final String assetPath,
   private Content putMetadata(final String assetPath,
                               final Content content,
                               final AssetKind assetKind,
+                              final String group,
                               final String project,
-                              final String version,
-                              final String group)
+                              final String version)
       throws IOException
   {
+    TempBlob updatedBlob = null;
     StorageFacet storageFacet = facet(StorageFacet.class);
     try (TempBlob tempBlob = storageFacet.createTempBlob(content.openInputStream(), HASH_ALGORITHMS)) {
       AttributesMap attributesMap;
-      TempBlob blob = tempBlob;
       switch (assetKind) {
         case DOWNLOAD_URL:
-          blob = absoluteUrlRemover.removeAbsoluteUrls(tempBlob, getRepository());
-          // break intentionally left out!
+          String indexAssetName = getProjectIndexName(group, project, version);
+          updatedBlob = absoluteUrlRemover.updateAbsoluteUrls(tempBlob, getRepository(), indexAssetName);
+          attributesMap = ConanManifest.parse(updatedBlob);
+          break;
         case CONAN_MANIFEST:
           attributesMap = ConanManifest.parse(tempBlob);
           break;
         case CONAN_FILE:
-          //TODO: Parse file using to get license information and description, email, author etc
-
+          //TODO: Parse file using to get license information and description, email, group etc
+          attributesMap = new AttributesMap();
+          break;
         default:
           attributesMap = new AttributesMap();
           break;
       }
-      return doSaveMetadata(assetPath, blob, content, assetKind, attributesMap, project, version, group);
+      return doSaveMetadata(assetPath, updatedBlob != null ? updatedBlob : tempBlob, content, assetKind, attributesMap, project, version, group);
+    } finally {
+      if(updatedBlob != null && DOWNLOAD_URL.equals(assetKind)) {
```

Comment:
Rather than do it this way I might just refactor this so that each `case` is effectively standalone, I'm not sure it's worth doing it this way in an age where we have autocloseable resources. I'd just write it so that we can have the updatedBlob exist within its own try-with-resources.

Reply:
https://github.com/sonatype-nexus-community/nexus-repository-conan/pull/6/commits/7bf9e697509d0b842dc0d45ecffea95fb3fb7610